ros2_graph/ros_element.py

```diff
@@ -0,0 +1,225 @@
+#! /usr/bin/env python3
+# Copyright 2023 Kiwicampus Inc.
+#
+# Licensed under GNU GENERAL PUBLIC, version 3 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.gnu.org/licenses/gpl-3.0.en.html
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import List, Dict
+
+
+class ElementType(Enum):
+    """!
+    Enumeration class for ROS element types
+    """
+
+    MAIN = 0
+    NODE = 1
+    TOPIC = 2
+    SERVICE = 3
+    ACTION = 4
+
+
+def element_style(element_type: ElementType) -> str:
+    """!
+    Just map element types and and it's string
+    """
+    if element_type == ElementType.MAIN:
+        return "main"
+    if element_type == ElementType.NODE:
+        return "node"
+    if element_type == ElementType.TOPIC:
+        return "topic"
+    if element_type == ElementType.SERVICE:
+        return "service"
+    if element_type == ElementType.ACTION:
+        return "action"
+    return "default"
+
```

Comment:
this could be just the `__str__` method of `ElementType`

Reply:
Done